index.js

```diff
@@ -762,6 +765,14 @@
     typeEq('sanctuary-def/Type')
   );
 
+  var VariadicType = NullaryType(
+    'sanctuary-def/VariadicType',
+    '',
+    function(x) {
+      return test([], Type, x) || test([], AnyFunction, x);
```

Comment:
So you vote for not combining `1` and `3`? In that case, shall I undo the diff that introduces `applyParameterizedType` (*singular*)?